### PR TITLE
Disable tracing when making calls to tcollector

### PR DESCRIPTION
**Pull request by prashantv**
Add methods to ContextBuilder to disable tracing, and use those in the ZipKin trace reporter.

r @junchaowu

### CODE REVIEW

File: trace/zipin_tracereporter_test.go [hunk 1]
```diff
@@ -35,12 +36,49 @@ import (
...
+	ch.Peers().Add(tcollectorCh.PeerInfo().HostPort)
+
+	called := make(chan int)
+	ret := &gen.Response{Ok: true}
+	mockServer.On("Submit", mock.Anything, mock.Anything).Return(ret, nil).Run(func(args mock.Arguments) {
+		called <- 1
+	})
+
+	// Make some calls, and validate that the trace reporter is not called recursively.
+	require.NoError(t, json.Register(tcollectorCh, json.Handlers{"op": func(ctx json.Context, arg map[string]interface{}) (map[string]interface{}, error) {
```
Comment (junchaowu) [inline trace/zipin_tracereporter_test.go]: why do you use json

Comment (prashantv) [inline trace/zipin_tracereporter_test.go]: Was the simplest to set up an inline server to listen on (just needed any listener)

File: context_builder.go [hunk 2]
```diff
@@ -119,6 +129,10 @@ func (cb *ContextBuilder) Build() (ContextWithHeaders, context.CancelFunc) {
 		timeout = defaultTimeout
 	}
 
+	if cb.TracingDisabled {
+		cb.span.EnableTracing(false)
```
Comment (junchaowu) [inline context_builder.go]: this function look weird. should be enableTracing / disableTracing

Comment (prashantv) [inline context_builder.go]: Yeah looks weird, but it's a single function to set true/false. It's not really worth breaking the public API for it, especially not as part of this change.

**Comment by prashantv** [comment]
Merging to unblock other changes.

**Pull request merged**